### PR TITLE
Added router support for optional path

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Added support for optional path in Router

### CODE REVIEW

**Comment by ricardo-a-alves-alb** [comment]
How would the route look like?

`player/:id?`

**Review by michielvandergeest — COMMENTED** [review]
LGTM - can we just change some `let`s into `const`s as discussed.

**Comment by frank-weindel** [comment]
Seems a bit fragile to me and would really benefit from documentation.

One might assume that a path like this should work:

```
'player/:assetId?/:playlistId'
```

But it would result in these two Route paths being used without any indication of error:

```
'player/:assetId'
'player/:assetId?'
```

**Comment by frank-weindel** [comment]
FWIW I found the library that Express uses for Route path parsing. Wondering if it might help: https://www.npmjs.com/package/path-to-regexp

**Comment by anandpatel0123** [comment]
Hi, when this is going to be rolled out? eagerly waiting.

**Comment by uguraslan** [comment]
The router plugin supports using regular expressions in route paths (as documented [here](https://lightningjs.io/docs/#/lightning-sdk-reference/plugins/router/configuration?id=using-regular-expressions-in-routes)) like:

```js
// this will match #player/1493847
// but will fail on #player/ah26134
{
    path: 'player/:playlistId${/[0-9]{3,8}/i}',
    component: Player
}
```

`?` is part of regex syntax, it can be even used as a literal character in regex (`\?`). The following example will fail due to the current optional path implementation.

```js
{
    path: 'player/:playlistId${/[0-9]{3,8}0?/i}',
    component: Player
}
```

File: src/Router/utils/router.js [hunk 1]
```diff
@@ -262,6 +262,22 @@ const init = config => {
       console.error(`[Router]: ${config.bootComponent} is not a valid boot component`)
     }
   }
+  config.routes.forEach(item => {
+    let path = item.path
+    // Pattern to identify the last path of the route
+    // It should start with "/:" + any word  and ends with "?"
+    // Example /player/:asset/:assetId?
+    const pattern = /\/:{1}\w+\?$/
```
Comment (uguraslan) [inline src/Router/utils/router.js:270]: `\w` matches `[A-Za-z0-9_]` character class, whereas as far as I see there is no restriction about path names in the Router docs. For example, they can include hyphen `-` or and other characters from other languages. Also `{1}` is not required when we try to match single literal character. So I think the following regex might be better:

```js
const pattern = /\/:.*?\?$/u
```

**Comment by uguraslan** [comment]
@michielvandergeest I think everything seems to be okay for this PR. Can you please review it again with the latest changes?

**Review by michielvandergeest — APPROVED** [review]
LGTM

**Pull request merged**